x-pack/plugins/security_solution/server/lib/detection_engine/index/delete_all_index.ts

```diff
@@ -4,14 +4,42 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { IndicesDeleteParams } from 'elasticsearch';
+import { IndicesDeleteParams, Client } from 'elasticsearch';
 import { CallWithRequest } from '../types';
 
+const MAX_ATTEMPTS = 5;
+
 export const deleteAllIndex = async (
-  callWithRequest: CallWithRequest<IndicesDeleteParams, boolean>,
-  index: string
+  callWithRequest: CallWithRequest<IndicesDeleteParams, ReturnType<Client['indices']['getAlias']>>,
+  pattern: string
 ): Promise<boolean> => {
-  return callWithRequest('indices.delete', {
-    index,
-  });
+  for (let attempt = 1; ; attempt++) {
+    if (attempt > MAX_ATTEMPTS) {
```

Comment:
For `MAX_ATTEMPTS`, could we just add it with a default to the function signature like so:

```ts
export const deleteAllIndex = async (
  callWithRequest: CallWithRequest<IndicesDeleteParams, ReturnType<Client['indices']['getAlias']>>,
  pattern: string,
  maxAttempts = 5,
): Promise<boolean> => {	): Promise<boolean> => {
```